src/alfasim_sdk/_internal/alfacase/case_description.py

```diff
@@ -2669,6 +2677,21 @@ def _validate_inversion_point_water_cut(self, attribute, value):
             and 0.0 <= value.GetValue("m3/m3") <= 1.0
         ), "Invalid inversion point water-cut"
 
+    @emulsion_relative_viscosity_tuning_factor.validator
+    def _validate_emulsion_relative_viscosity_tuning_factor(self, attribute, value):
+        assert isinstance(value, Curve), "Invalid tuning factor curve"
```

Comment:
`attrib_curve` already guarantee that the value is a curve, no domain/image checking though I think this could qualify as a valid addition to `attrib_curve` functionality.

The range checking is really better de implemented case by case in my opinion.
Of course if this is done ostensibly (I can't remember how often it is done) probably we should add that feature to `attrib_curve`.

#248

Reply:
> attrib_curve already guarantee that the value is a curve
Good to know

>The range checking is really better de implemented case by case in my opinion.
Of course if this is done ostensibly (I can't remember how often it is done) probably we should add that feature to attrib_curve.

In this case, it is better indeed to check the range in the class where the curve is defined since a curve itself is a very generic concept